server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/aggregators/CountValueAggregator.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.index.compositeindex.datacube.startree.aggregators;
+
+import org.opensearch.index.compositeindex.datacube.MetricStat;
+import org.opensearch.index.compositeindex.datacube.startree.aggregators.numerictype.StarTreeNumericType;
+
+/**
+ * Count value aggregator for star tree
+ *
+ * @opensearch.experimental
+ */
+public class CountValueAggregator implements ValueAggregator<Long> {
+    public static final StarTreeNumericType VALUE_AGGREGATOR_TYPE = StarTreeNumericType.LONG;
+
+    @Override
+    public MetricStat getAggregationType() {
+        return MetricStat.COUNT;
+    }
+
+    @Override
+    public StarTreeNumericType getAggregatedValueType() {
+        return VALUE_AGGREGATOR_TYPE;
+    }
+
+    @Override
+    public Long getInitialAggregatedValueForSegmentDocValue(Long segmentDocValue, StarTreeNumericType starTreeNumericType) {
+        return 1L;
```

Comment:
I'm not fully sure what you are suggesting. Can you please elaborate?

Reply:
https://github.com/opensearch-project/OpenSearch/blob/6b8b3efe01a62c221f308a2e3b019d75a7f5ad8a/server/src/main/java/org/opensearch/index/mapper/DocCountFieldMapper.java#L81

Can be done later but wondering if this fits the use case

Reply:
I am still not fully sure if I get this. They are returning 1L if the value is empty else the converted value

Reply:
https://opensearch.org/docs/2.4/opensearch/bucket-agg/#account-for-pre-aggregated-data - check this , basically users can also ingest pre-aggregated data. 

Instead of ingesting , say `1,1,2` 100 times, we have an option to ingest `1,1,2` with `_doc_count` as 100.

Reply:
Lets create an issue for this

Reply:
Issue: https://github.com/opensearch-project/OpenSearch/issues/14767